src/components/Table/index.tsx

```diff
@@ -0,0 +1,302 @@
+					if (event.target.checked) {
+						checkedRowsMap.set(row[rowKey], row);
+					} else {
+						checkedRowsMap.delete(row[rowKey]);
+					}
+					lastSelected.current = row;
+				}
+
+				const filteredArray = Array.from(checkedRowsMap.values());
+				onCheck?.(filteredArray, filteredArray.length > 0 ? 'some' : 'none');
+			};
+		},
+		[visibleRowsMap, checkedRowsMap, rowKey, onCheck],
+	);
+
+	return (
+		<Box sx={{ width: '100%' }}>
+			<Paper sx={{ width: '100%', mb: 2 }}>
+				<TableToolbar
+					numSelected={numSelected}
+					columns={columns}
+					actions={actions}
+					onColumnPreferencesChange={onColumnPreferencesChange}
+				/>
+				<TableContainer sx={{ maxHeight: '60vh' }}>
+					<StyledMaterialTable stickyHeader>
+						<TableHead
+							data-display="table-head"
+							columns={columns}
+							numSelected={numSelected}
+							checkedState={checkedState}
+							order={sort.direction}
+							orderBy={sort.key}
+							onSelectAllClick={onCheck}
+							onRequestSort={handleOnSort}
+							rowCount={totalItems}
+						/>
+						<TableBody data-display="table-body">
+							{visibleRows.map((row, index) => {
+								const labelId = `enhanced-table-checkbox-${index}`;
+								const checked = isChecked(row);
+
+								return (
+									<TableRow
+										hover
+										role="checkbox"
+										data-display="table-row"
+										aria-checked={false}
+										tabIndex={-1}
+										key={row[rowKey] as string}
+										selected={false}
+										sx={{
+											cursor: 'pointer',
+											background:
+												index % 2 === 0
+													? '#f3f5f7!important'
+													: '#fff!important',
+											'&:hover': {
+												background: '#e6f2fc!important',
+											},
+										}}
+									>
+										<TableCell
+											data-display="table-cell"
+											data-table="table_cel__sticky"
+											padding="checkbox"
+											sx={{
+												backgroundColor: 'inherit',
+												whiteSpace: 'nowrap',
+											}}
+										>
+											<Checkbox
+												color="primary"
+												onClick={handleToggleCheck(row)}
+												checked={checkedState === 'all' || checked}
+												inputProps={{
+													'aria-labelledby': labelId,
+												}}
+											/>
+										</TableCell>
+										{columns.map((c) => {
+											return c.selected ? (
+												<TableCell
+													data-display="table-cell"
+													data-key={row[rowKey]}
+													component="th"
+													id={labelId}
+													scope="row"
+													padding="none"
+													sx={{ whiteSpace: 'nowrap' }}
+													onClick={(event) => {
+														onRowClick?.(row, event);
+													}}
+												>
+													{c.render(row[c.field], row)}
+												</TableCell>
+											) : null;
+										})}
+									</TableRow>
+								);
+							})}
+						</TableBody>
+					</StyledMaterialTable>
+				</TableContainer>
+				<Divider m={0} />
```

Comment:
Why do we opt for this instead of just a css border top on the `TablePagination`? Or a css border bottom on the `TableContainer`?